modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryIndexing.java

```diff
@@ -356,6 +356,14 @@ public void remove(GridCacheContext cctx, GridQueryTypeDescriptor type, CacheDat
      */
     @Nullable IgniteInternalFuture<?> rebuildIndexesFromHash(GridCacheContext cctx, boolean force);
 
+    /**
+     * Rebuild text index for the given cache if necessary.
+     *
+     * @param cctx Cache context.
+     * @return Future completed when index rebuild finished.
+     */
+    @Nullable IgniteInternalFuture<?> rebuildTextIndexes(GridCacheContext cctx);
```

Comment:
```suggestion
    @Nullable IgniteInternalFuture<?> rebuildInMemoryIndexes(GridCacheContext cctx);
```
Maybe we can extend this on all in-memory indexes?

Reply:
Renamed